pandahub/lib/PandaHub.py

```diff
@@ -665,17 +673,30 @@ def get_subnet_from_db_by_id(self, net_id, bus_filter=None, include_results=True
         self._add_element_from_collection(net, db, "switch", net_id, switch_filter,
                                           geo_mode=geo_mode, variants=variants, dtypes=dtypes)
 
+        substations_in_net = db["net_bus"].find({"net_id": net_id, "index": {"$in": buses}}, {"_id": 0, "substation": 1}).distinct("substation")
+        substation_filter = {"index": {"$in": substations_in_net}}
+        self._add_element_from_collection(net, db, "substation", net_id, substation_filter,
+                                          geo_mode=geo_mode, variants=variants, dtypes=dtypes)
+
         # add node elements
         node_elements = ["load", "sgen", "gen", "ext_grid", "shunt", "xward", "ward", "motor", "storage"]
         branch_elements = ["trafo", "line", "trafo3w", "switch", "impedance"]
-        all_elements = node_elements + branch_elements + ["bus"]
+        all_elements = node_elements + branch_elements + ["bus", "substation"] + list(additional_filters.keys())
         all_elements = list(set(all_elements) - set(ignore_elements))
 
+        # Add elements for which the user has provided a filter function
+        for element, filter_func in additional_filters.items():
```

Comment:
@jthurner Should we add a check here (or somewhere in this function) that `additional_filters` does not contain an element out of `node_elements` or `branch_elements`? Otherwise, these might be handled twice in the current implementation, nullifying the desired performance boost. If so, how should this work? Should the user be able to override the filters for elements in `node_elements` and `branch_elements` or should this raise an error?